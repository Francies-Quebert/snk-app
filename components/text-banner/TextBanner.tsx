import { View, StyleSheet } from "react-native";
import React from "react";
import Text from "../custom-text/Text";
import { theme } from "@/styles/theme";

type TextBannerProps = {
  text: string;
  codeText: string;
  linkText: string;
};

const TextBanner = ({ text, codeText, linkText }: TextBannerProps) => {
  return (
    <View style={styles.container}>
      <Text color="white" style={styles.textContainer}>
        {text}
        <>
          {" "}
          <Text weight="bold" color="white">
            {codeText}
          </Text>
          {" - "}
          <Text color="white" weight="thin" style={styles.linkText}>
            {linkText}
          </Text>
        </>
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: theme.colors.pink,
    paddingVertical: theme.spacing.small,
    paddingHorizontal: theme.spacing.medium,
  },
  textContainer: {
    textAlign: "center",
    alignSelf: "center",
  },
  linkText: { textDecorationLine: "underline" },
});
export default React.memo(TextBanner);
