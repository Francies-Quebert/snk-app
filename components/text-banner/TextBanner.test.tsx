import React from "react";
import { render } from "@testing-library/react-native";
import TextBanner from "./TextBanner";

describe("TextBanner Component", () => {
  it("renders the text, codeText, and linkText correctly", () => {
    const props = {
      text: "Sample text",
      codeText: "Code example",
      linkText: "Click here",
    };

    const { getByText } = render(<TextBanner {...props} />);

    expect(getByText(`${props.text} ${props.codeText} - ${props.linkText}`)).toBeOnTheScreen();
  });

  it("matches the snapshot", () => {
    const props = {
      text: "Snapshot text",
      codeText: "Snapshot code",
      linkText: "Snapshot link",
    };

    const { toJSON } = render(<TextBanner {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
