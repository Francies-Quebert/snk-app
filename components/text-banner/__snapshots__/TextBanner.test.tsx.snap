// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBanner Component matches the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#CA9396",
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "FuturaforSNK",
          "fontSize": 16,
          "fontWeight": "400",
        },
        {
          "alignSelf": "center",
          "textAlign": "center",
        },
      ]
    }
  >
    Snapshot text
     
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "FuturaforSNK",
            "fontSize": 16,
            "fontWeight": "700",
          },
          undefined,
        ]
      }
    >
      Snapshot code
    </Text>
     - 
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "FuturaSerieBQThin",
            "fontSize": 16,
            "fontWeight": "100",
          },
          {
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      Snapshot link
    </Text>
  </Text>
</View>
`;
